rskj-core/src/main/java/org/ethereum/core/BlockFactory.java

```diff
@@ -62,7 +103,7 @@ public Block decodeBlock(byte[] rawData) {
 
     private Block decodeBlock(byte[] rawData, boolean sealed) {
         RLPList block = RLP.decodeList(rawData);
-        if (block.size() != 3) {
+        if (block.size() != NUMBER_OF_EXTRA_HEADER_FIELDS) {
```

Comment:
Not sure you should be comparing against this constant.

If I understood correctly `NUMBER_OF_EXTRA_HEADER_FIELDS` refers to version, egdes and base event. The block size refers to headers, transactions and uncles. Same value but too different concepts the way I see it